crates/core/src/metadata/manager.rs

```diff
@@ -45,6 +50,16 @@ use crate::task_center;
 pub(super) type CommandSender = mpsc::UnboundedSender<Command>;
 pub(super) type CommandReceiver = mpsc::UnboundedReceiver<Command>;
 
+/// Metadata requirements for each role. Unless a mask is specified, implicitly all metadata will be synced.
+/// This mechanism allows for nodes to opt out of syncing metadata that they don't need.
+pub static ROLE_METADATA_REQUIREMENTS: LazyLock<HashMap<Role, EnumSet<MetadataKind>>> =
+    LazyLock::new(|| {
```

Comment:
Can't use the EnumSet builder in a static context, hence the LazyLock. This is an optional optimisation though, we can also drop it for now.